src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveILGeneratorTests.cs

```diff
@@ -2174,5 +2174,39 @@ public void RecursiveSumTest()
                 Assert.Equal(55, recursiveMethodFromDisk.Invoke(null, [10]));
             }
         }
+
+        [Fact]
+        public void CallOpenGenericMembersFromConstructedGenericType()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo saveMethod);
+                MethodBuilder method = type.DefineMethod("M1", MethodAttributes.Public, null, null);
+
+                ILGenerator ilGenerator = method.GetILGenerator();
+                LocalBuilder span = ilGenerator.DeclareLocal(typeof(ReadOnlySpan<char>));
+                LocalBuilder str = ilGenerator.DeclareLocal(typeof(string));
+
+                ilGenerator.Emit(OpCodes.Ldstr, "hello");
+                ilGenerator.Emit(OpCodes.Call, typeof(MemoryExtensions).GetMethod("AsSpan", [typeof(string)])!);
+                ilGenerator.Emit(OpCodes.Stloc, span);
+                ilGenerator.Emit(OpCodes.Ldloca_S, span);
+                ilGenerator.Emit(OpCodes.Ldc_I4_1);
+                ilGenerator.Emit(OpCodes.Call, typeof(ReadOnlySpan<char>).GetMethod("Slice", [typeof(int)])!);
+                ilGenerator.Emit(OpCodes.Stloc, span);
+                ilGenerator.Emit(OpCodes.Ldloca_S, span);
+                ilGenerator.Emit(OpCodes.Constrained, typeof(ReadOnlySpan<char>));
+                ilGenerator.Emit(OpCodes.Callvirt, typeof(object).GetMethod("ToString"));
+                ilGenerator.Emit(OpCodes.Stloc, str);
+                ilGenerator.EmitWriteLine(str);
```

Comment:
We shouldn't actually WriteLine unless we're going to validate that stdout gets what we exoect it to

Reply:
I can return the string for validation, as don't know how to check what stdout gets.

Reply:
> I can return the string for validation

That's fine.